tests/functional_tests/functional_tests.py

```diff
@@ -53,11 +54,62 @@ def test_serializing(self):
 K8zMyhluSZuJtSTvdZopmDkTYjOGpmMI9vWcK'
         )
 
+    def test_serializing_with_binary_v1(self):
+        identifier = base64.b64decode(u'AK2o+q0Aq9+bONkXw7ky7HAuhCLO9hhaMMc==')
+        m = Macaroon(
+            location='http://mybank/',
+            identifier=identifier,
+            key='this is our super secret key; only we should know it',
+            version=MACAROON_V1
+        )
+        m.add_first_party_caveat('test = caveat')
+        n = Macaroon.deserialize(m.serialize())
+        assert_equal(m.identifier, n.identifier)
+        assert_equal(m.version, n.version)
+
+    def test_serializing_with_binary_v2(self):
+        identifier = base64.b64decode('AK2o+q0Aq9+bONkXw7ky7HAuhCLO9hhaMMc==')
+        m = Macaroon(
+            location='http://mybank/',
+            identifier=identifier,
+            key='this is our super secret key; only we should know it',
+            version=MACAROON_V2
+        )
+        m.add_first_party_caveat('test = caveat')
+        n = Macaroon.deserialize(m.serialize())
+        assert_equal(m.identifier, n.identifier)
+        assert_equal(m.version, n.version)
+
+    def test_serializing_v1(self):
+        m = Macaroon(
+            location='http://mybank/',
+            identifier='we used our secret key',
+            key='this is our super secret key; only we should know it',
+            version=MACAROON_V1
+        )
+        m.add_first_party_caveat('test = caveat')
+        n = Macaroon.deserialize(m.serialize())
+        assert_equal(m.identifier, n.identifier)
+        assert_equal(m.version, n.version)
+
+    def test_serializing_v2(self):
+        m = Macaroon(
+            location='http://mybank/',
+            identifier='we used our secret key',
+            key='this is our super secret key; only we should know it',
+            version=MACAROON_V2
+        )
+        m.add_first_party_caveat('test = caveat')
+        n = Macaroon.deserialize(m.serialize())
+        assert_equal(m.identifier, n.identifier)
+        assert_equal(m.version, n.version)
+
     def test_serializing_strips_padding(self):
         m = Macaroon(
             location='http://mybank/',
             identifier='we used our secret key',
-            key='this is our super secret key; only we should know it'
+            key='this is our super secret key; only we should know it',
+            version=MACAROON_V1
```

Comment:
I've been meaning to update the tests to use pytest, I'd like to parameterize these over macaroon versions. I can make a followup PR for that.